firmware/boron-ins-fsm/test/mocks/Particle.h

```diff
@@ -48,4 +62,51 @@ class MockParticle
     }
 };
 extern String fullPublishString;
-extern MockParticle Particle;
\ No newline at end of file
+extern MockParticle Particle;
+
+typedef enum LogLevel {
+    LOG_LEVEL_ALL = 1, // Log all messages
+    LOG_LEVEL_TRACE = 1,
+    LOG_LEVEL_INFO = 30,
+    LOG_LEVEL_WARN = 40,
+    LOG_LEVEL_ERROR = 50,
+    LOG_LEVEL_PANIC = 60,
+    LOG_LEVEL_NONE = 70 // Do not log any messages
+} LogLevel;
+
+class MockLogger
+{
+public:
+    MockLogger() {}
+
+public:
+    void warn(const char *fmt, ...) const __attribute__((format(printf, 2, 3))) {
+        printf("LOG.WARN\n");
+    } 
+
+    void error(const char *fmt, ...) const __attribute__((format(printf, 2, 3))) {
+        printf("LOG.ERROR\n");
+    }
+
+    void info(const char *fmt, ...) const __attribute__((format(printf, 2, 3))) {
+        printf("LOG.INFO\n");
+    }
+};
+extern MockLogger Log;
+
+uint32_t millis() {
+    struct timespec ts;
+
+#ifdef __linux__
+    clock_gettime(CLOCK_MONOTONIC, &ts);
+#elif _WIN32
+    clock_gettime_monotonic(&ts);
+#else
+#endif
+
+    return (uint32_t) (uint64_t)(ts.tv_nsec / 1000000) + ((uint64_t)ts.tv_sec * 1000ull);
+}
+
+void delay(unsigned long ms) {
+    return;
+}
```

Comment:
Are these functions that are defined in the real Particle.h file?

Reply:
They are defined in spark_wiring_ticks.h - I moved both functions to a file called mock_ticks.h